app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -1282,68 +1264,51 @@ class BrowserTabFragment : Fragment(), FindListener, CoroutineScope, DaxDialogLi
     }
 
     inner class BrowserTabFragmentDecorator {
-        fun decorateToolbar(viewState: BrowserViewState) {
-            if (!variantManager.getVariant().hasFeature(VariantManager.VariantFeature.BottomBarNavigation)) {
-                decorator.decorateToolbarActions(viewState)
-            }
-        }
 
-        private fun decorateToolbarActions(viewState: BrowserViewState) {
-            tabsButton?.isVisible = viewState.showTabsButton
-            fireMenuButton?.isVisible = viewState.showFireButton
-            menuButton?.isVisible = viewState.showMenuButton
+        fun decorate() {
+            decorateToolbarWithButtons()
+            createPopupMenu()
+            configureShowTabSwitcherListener()
+            configureLongClickOpensNewTabListener()
         }
 
-        fun decorateWithToolbar() {
-            hideBottomBar()
-            decorateAppBarWithToolbarOnly()
-            createPopupMenuWithToolbarOnly()
-            configureShowTabSwitcherListenerWithToolbarOnly()
-            configureLongClickOpensNewTabListenerWithToolbarOnly()
-            removeUnnecessaryLayoutBehaviour()
+        fun decorateToolbarMenus(viewState: BrowserViewState) {
+            decorator.updateToolbarActionsVisibility(viewState)
```

Comment:
replace method call with `updateToolbarActionsVisibility` implementation? Also to simplify method forwarding for just one line.